test/auditing.jl

```diff
@@ -517,6 +517,49 @@ end
     end
 end
 
+@testset "Auditor - execution permission" begin
+    mktempdir() do build_path
+        build_output_meta = nothing
+        product = LibraryProduct("libfoo", :libfoo)
+        @test_logs (:info, r"Making .*libfoo.* executable") match_mode=:any begin
+            build_output_meta = autobuild(
+                build_path,
+                "exec",
+                v"1.0.0",
+                # No sources
+                FileSource[],
+                # Build a library without execution permissions
+                raw"""
+                mkdir -p "${libdir}"
+                cc -o "${libdir}/libfoo.${dlext}" -shared /usr/share/testsuite/c/dyn_link/libfoo/libfoo.c
+                chmod 640 "${libdir}/libfoo.${dlext}"
+                """,
+                # Build for our platform
+                [platform],
+                # Ensure our library product is built
+                [product],
+                # No dependencies
+                Dependency[];
+                verbose = true,
+                require_license = false
+            )
+        end
+
+        # Extract our platform's build
+        @test haskey(build_output_meta, platform)
+        tarball_path, tarball_hash = build_output_meta[platform][1:2]
+        @test isfile(tarball_path)
+
+        # Unpack it somewhere else
+        @test verify(tarball_path, tarball_hash)
+        testdir = joinpath(build_path, "testdir")
+        mkdir(testdir)
+        unpack(tarball_path, testdir)
+        libfoo_path = joinpath(testdir, build_output_meta[platform][4][product]["path"])
+        @test_broken filemode(libfoo_path) & 0o777 == 0o750
```

Comment:
I believe we have an issue when packing the tarball: the file has permissions `0x750` as expected right after `ensure_executability`, but in the tarball it becomes `0o755`.  However this is orthogonal to this PR

Reply:
It seems to be a deliberate decision in `Tar.jl`:

> files that are executable by the owner are archived with mode `0o755`;

I can reproduce this with
```julia
julia> using Tar

julia> mktempdir() do dir
           # Create the root of the tarball
           root = joinpath(dir, "root")
           mkpath(root)
           # Create the file and give permissions 750
           path = joinpath(root, "testfile")
           touch(path)
           chmod(path, 0o750)
           # Package the root
           tarball = Tar.create(root, joinpath(dir, "test.tar"))
           files = Tar.list(tarball)
           # Get the permissions of the file
           string(filter(f -> f.type === :file, files)[1].mode; base=8)
       end
"755"
```

Reply:
Yeah; I think it’s good for BB to be correct, and Tar is fine to do it’s thing as well.